diff -urN gsmartcontrol-0.8.6/src/hz/sync_policy_glib.h gsmartcontrol-0.8.6-patched/src/hz/sync_policy_glib.h
--- gsmartcontrol-0.8.6/src/hz/sync_policy_glib.h	2011-03-06 19:21:07.000000000 +1000
+++ gsmartcontrol-0.8.6-patched/src/hz/sync_policy_glib.h	2011-11-03 02:19:45.689866738 +1100
@@ -21,16 +21,8 @@
 // The warnings are completely harmless and visible on some versions of glib only.
 // However, due to their number, I decided to implement this workaround.
 
-#ifndef _WIN32
-	// same as stock version, but an additional cast to (void*) is added.
-	#define hz_glib_static_mutex_get_mutex(mutex) \
-		( g_thread_use_default_impl ? ((GMutex*) ((void*)((mutex)->static_mutex.pad))) : \
-		g_static_mutex_get_mutex_impl_shortcut(&((mutex)->runtime_mutex)) )
-
-#else
-	// win32 has different definition of this macro, so default to stock version.
-	#define hz_glib_static_mutex_get_mutex(mutex) g_static_mutex_get_mutex(mutex)
-#endif
+//GLib API changed in 2.31 and the workaround doesn't work anymore, so use stock version
+#define hz_glib_static_mutex_get_mutex(mutex) g_static_mutex_get_mutex(mutex)
 
 
 #define hz_glib_static_mutex_lock(mutex) \
